### PR TITLE
Changes to LogitItem base class

### PR DESCRIPTION
**Pull request by L-E-son**
Added an additional abstract method to validate the input given.

### CODE REVIEW

File: Breadboard/Logic/Gates/GateAnd.cs [hunk 1]
```diff
@@ -8,25 +8,28 @@ namespace Breadboard.Logic.Gates
...
+        protected override bool Evaluate(bool[] inputs)
         {
-            if (inputs.Length < 2)
-            {
-                throw new ArgumentException(string.Format("AND Gate expects 2+ arguments, but received {0} arguments", inputs.Length));
-            }
-            else
+            Validate(inputs);
+
+            for (int i = 0; i < inputs.Length; i++)
```
Comment (Keahi) [inline Breadboard/Logic/Gates/GateAnd.cs]: We can replace this check by using `return !inputs.Contains(false)`

File: Breadboard/Logic/Gates/GateOr.cs [hunk 2]
```diff
@@ -6,28 +6,26 @@
...
-                }
+            Validate(inputs);
 
-                return false;
-            }
+            bool result = false;
+
+            foreach (var input in inputs)
+                result |= input;
+
```
Comment (Keahi) [inline Breadboard/Logic/Gates/GateOr.cs]: 👍

Comment (Keahi) [reply Breadboard/Logic/Gates/GateOr.cs]: No change here I just thought it was good

File: Breadboard/Logic/Gates/GateAnd.cs (hunk 1 above)
Comment (L-E-son) [reply Breadboard/Logic/Gates/GateAnd.cs]: lol, I just left that one the same. I'm going to update it.

**Comment by L-E-son** [comment]
Updated!

**Pull request merged**